core/services/log/broadcaster.go

```diff
@@ -365,18 +365,30 @@ func (b *broadcaster) onNewHeads() {
 			keptDepth = 0
 		}
 
-		logs, minBlockNum := b.logPool.getLogsToSend(latestBlockNum)
-
-		if len(logs) > 0 {
-			broadcasts, err := b.orm.FindConsumedLogs(minBlockNum, latestBlockNum)
-			if err != nil {
-				logger.Errorf("Failed to query for log broadcasts, %v", err)
-				return
+		if b.registrations.lowestNumConfirmations == 0 && b.registrations.highestNumConfirmations == 0 {
```

Comment:
Is this an optimization or a necessary special case?

Reply:
necessary - we can't check against the latest block number so have to pick all logs from the pool to be able to send them to these subscribers